File: Makecode_minecraft_farm/farm.py
setpos: Position = None
linemove = 0

def item_interacted_clock():
    gameplay.set_weather(CLEAR)
    gameplay.time_set(gameplay.time(DAY))
player.on_item_interacted(CLOCK, item_interacted_clock)

def item_interacted_wooden_shovel():
    blocks.fill(AIR, pos(-10, 0, 0), pos(10, 0, 10), FillOperation.REPLACE)
    blocks.fill(GRASS,
        pos(-10, -1, 0),
        pos(10, -1, 10),
        FillOperation.REPLACE)
player.on_item_interacted(WOODEN_SHOVEL, item_interacted_wooden_shovel)

def item_interacted_carrot():
    agent.set_item(CARROTS, 64, 1)
    agent.teleport_to_player()
    agent.move(FORWARD, 1)
    setpos = agent.get_position()
    linemove = 1
    for index in range(5):
        for index2 in range(5):
            agent.till(FORWARD)
            agent.move(FORWARD, 1)
            agent.set_slot(1)
            agent.place(DOWN)
        agent.teleport_to_player()
        agent.turn(LEFT_TURN)
        agent.move(FORWARD, linemove)
        linemove += 1
        agent.turn(RIGHT_TURN)
        agent.move(FORWARD, 1)
    agent.teleport_to_player()
    agent.turn(LEFT_TURN)
    agent.move(FORWARD, 2)
    agent.turn(RIGHT_TURN)
    agent.move(FORWARD, 4)
    agent.destroy(DOWN)
    agent.move(DOWN, 1)
    blocks.place(WATER, agent.get_position())
    agent.set_item(LOG_OAK, 64, 27)
    agent.set_item(OAK_FENCE, 64, 26)
    agent.collect_all()
    agent.teleport_to_player()
    agent.turn(RIGHT_TURN)
    agent.move(FORWARD, 1)
    agent.turn(LEFT_TURN)
    agent.move(FORWARD, 1)
    agent.destroy(DOWN)
    agent.set_slot(27)
    agent.place(DOWN)
    for index3 in range(3):
        for index4 in range(6):
            agent.move(FORWARD, 1)
            agent.destroy(DOWN)
            agent.collect_all()
            agent.set_slot(27)
            agent.place(DOWN)
            agent.set_slot(26)
            agent.place(BACK)
        agent.turn(LEFT_TURN)
    agent.move(FORWARD, 1)
    agent.place(BACK)
    agent.teleport_to_player()
player.on_item_interacted(CARROT, item_interacted_carrot)

def on_chat():
    agent.teleport_to_player()
player.on_chat("come", on_chat)

def item_interacted_seeds():
    agent.set_item(CARROTS, 64, 1)
    agent.teleport_to_player()
    linemove = 1
    for index5 in range(5):
        for index6 in range(5):
            agent.move(FORWARD, 1)
            agent.set_slot(1)
            agent.place(FORWARD)
        agent.teleport_to_player()
        agent.turn(LEFT_TURN)
        agent.move(FORWARD, linemove)
        linemove += 1
        agent.turn(RIGHT_TURN)
    agent.teleport_to_player()
player.on_item_interacted(SEEDS, item_interacted_seeds)

def on_chat2():
    gameplay.set_weather(CLEAR)
    gameplay.time_set(gameplay.time(DAY))
player.on_chat("clr", on_chat2)

def item_interacted_stone_hoe():
    agent.teleport_to_player()
    setpos = agent.get_position()
    linemove = 1
    agent.turn(LEFT_TURN)
    for index7 in range(5):
        for index8 in range(5):
            agent.move(FORWARD, 1)
            agent.destroy(FORWARD)
            agent.collect(CARROT)
            agent.collect(POTATO)
        agent.teleport_to_player()
        agent.turn(LEFT_TURN)
        agent.move(FORWARD, linemove)
        linemove += 1
        agent.turn(RIGHT_TURN)
    agent.teleport_to_player()
player.on_item_interacted(STONE_HOE, item_interacted_stone_hoe)

File: Makecode_minecraft_farm/farm.js
let setpos: Position = null
let linemove = 0

player.onItemInteracted(CLOCK, function () {
    gameplay.setWeather(CLEAR)
    gameplay.timeSet(gameplay.time(DAY))
})
player.onItemInteracted(WOODEN_SHOVEL, function () {
    blocks.fill(
    AIR,
    pos(-10, 0, 0),
    pos(10, 0, 10),
    FillOperation.Replace
    )
    blocks.fill(
    GRASS,
    pos(-10, -1, 0),
    pos(10, -1, 10),
    FillOperation.Replace
    )
})
player.onItemInteracted(CARROT, function () {
    agent.setItem(CARROTS, 64, 1)
    agent.teleportToPlayer()
    agent.move(FORWARD, 1)
    setpos = agent.getPosition()
    linemove = 1
    for (let index = 0; index < 5; index++) {
        for (let index = 0; index < 5; index++) {
            agent.till(FORWARD)
            agent.move(FORWARD, 1)
            agent.setSlot(1)
            agent.place(DOWN)
        }
        agent.teleportToPlayer()
        agent.turn(LEFT_TURN)
        agent.move(FORWARD, linemove)
        linemove += 1
        agent.turn(RIGHT_TURN)
        agent.move(FORWARD, 1)
    }
    agent.teleportToPlayer()
    agent.turn(LEFT_TURN)
    agent.move(FORWARD, 2)
    agent.turn(RIGHT_TURN)
    agent.move(FORWARD, 4)
    agent.destroy(DOWN)
    agent.move(DOWN, 1)
    blocks.place(WATER, agent.getPosition())
    agent.setItem(LOG_OAK, 64, 27)
    agent.setItem(OAK_FENCE, 64, 26)
    agent.collectAll()
    agent.teleportToPlayer()
    agent.turn(RIGHT_TURN)
    agent.move(FORWARD, 1)
    agent.turn(LEFT_TURN)
    agent.move(FORWARD, 1)
    agent.destroy(DOWN)
    agent.setSlot(27)
    agent.place(DOWN)
    for (let index = 0; index < 3; index++) {
        for (let index = 0; index < 6; index++) {
            agent.move(FORWARD, 1)
            agent.destroy(DOWN)
            agent.collectAll()
            agent.setSlot(27)
            agent.place(DOWN)
            agent.setSlot(26)
            agent.place(BACK)
        }
        agent.turn(LEFT_TURN)
    }
    agent.move(FORWARD, 1)
    agent.place(BACK)
    agent.teleportToPlayer()
})
player.onChat("come", function () {
    agent.teleportToPlayer()
})
player.onItemInteracted(SEEDS, function () {
    agent.setItem(CARROTS, 64, 1)
    agent.teleportToPlayer()
    linemove = 1
    for (let index = 0; index < 5; index++) {
        for (let index = 0; index < 5; index++) {
            agent.move(FORWARD, 1)
            agent.setSlot(1)
            agent.place(FORWARD)
        }
        agent.teleportToPlayer()
        agent.turn(LEFT_TURN)
        agent.move(FORWARD, linemove)
        linemove += 1
        agent.turn(RIGHT_TURN)
    }
    agent.teleportToPlayer()
})
player.onChat("clr", function () {
    gameplay.setWeather(CLEAR)
    gameplay.timeSet(gameplay.time(DAY))
})
player.onItemInteracted(STONE_HOE, function () {
    agent.teleportToPlayer()
    setpos = agent.getPosition()
    linemove = 1
    agent.turn(LEFT_TURN)
    for (let index = 0; index < 5; index++) {
        for (let index = 0; index < 5; index++) {
            agent.move(FORWARD, 1)
            agent.destroy(FORWARD)
            agent.collect(CARROT)
            agent.collect(POTATO)
        }
        agent.teleportToPlayer()
        agent.turn(LEFT_TURN)
        agent.move(FORWARD, linemove)
        linemove += 1
        agent.turn(RIGHT_TURN)
    }
    agent.teleportToPlayer()
})
